main.py

```diff
@@ -55,7 +56,8 @@ def get(self, image_name):
         # FIXME : endpoint not ready
 
     def put(self, image_name):
-        new_image = create_base_image(driver)
+        body = request.json
+        new_image = create_base_image(driver, body)
```

Comment:
Passing the entire body of a http request to the underlying function is usually a bad idea, because there is no guarantee that the client included all the necessary information in that body, nor that the client didn't add additional information (which leads to security vulnerabilities if there are other weaknesses down the line).
You should get whatever data you need out of the body dict and pass it as independent (named?) arguments to the create_base_image function. The same applies to create_or_restore_instance.

To do this you'd have to do the same when running create_xxx as a script: parse the arguments in the main function and pass them to the function one by one.

Reply:
I changed it to passing the json body with ** to make it a variable lengthed dict, in case it has a parameter that doesn't exist in the function header it sends an error, otherwise it updates the default value.